### ISSUE TITLE
Huge Memory Leak 

### ISSUE THREAD

**Issue by iDhanush** [EverythingSuckz/TG-FileStreamBot#109]
In a 64GB Dedicated Server i ran The Bot For 7h with 4 clients

After 2 hours uptime the memory usage was about 5GB with a total of 400 workload (sum of all workloads of bot in approx)

After 7 hours of uptime the memory usage was a whopping 16GB with just total load of less than 70 

So we can say that the memory is not getting released even a bit even when load became 1/5th of the initial load

I whould really like to see an end to this 🥲, idk if the FileStream code or if the pyrogram itself is leaking memory but either way it's clear that there is a memory leak and i think it's from the media_sessions cuz the size of media_sesssion had also respectively without going down again 

if u need any more info or want to test any patches I'll be there for u xd

![usageStatsScreenShot](https://user-images.githubusercontent.com/119786916/235039635-28762f1c-8cf8-4e17-a121-2681bc677f93.jpg)

**Comment by EverythingSuckz** [comment]
Yeah, I'm aware of this from this [post](https://github.com/EverythingSuckz/TG-FileStreamBot/issues/108#issuecomment-1522077668) and it seems like I'll have to profile the whole code to see where all the memory is being used. Your `USE_SESSION_FILE` var is set to `True`, right?

**Comment by iDhanush** [comment]
Yep

**Comment by hyper-ub** [comment]
Mine works fine ig

**Comment by pdisk** [comment]
The Workloads are not getting released correctly
The finally block in yield file which is supposed to release the work load when it's done streaming doesn't get closed if the client seeks the video hence while streaming a new client will be assigned each time which makes even more ram consumption cuz the media session of each client now has to cache that file

In other words if the client seek the video 4 times it gets handled by 4 clients and create session and cache for each of those files in media session of each clients hence increasing the ram consumption

**Comment by FelinaHD** [comment]
ho we can solve this problem?

**Comment by Ygohel18** [comment]
I have used daemon service and timer to auto stop and start again every 3 hour? does it work?

**Comment by EverythingSuckz** [comment]
> I have used daemon service and timer to auto stop and start again every 3 hour? does it work?
> 

I'm working on a fix. I'll update it soon

**Comment by Netflixyapp** [comment]
Looking forward to this fix 🤞
